src/Microsoft.Health.Dicom.SqlServer/Features/ExtendedQueryTag/SqlExtendedQueryTagStoreV2.cs

```diff
@@ -94,23 +94,30 @@ public override async Task<IReadOnlyList<ExtendedQueryTagStoreEntry>> GetExtende
                 VLatest.GetExtendedQueryTag.PopulateCommand(sqlCommandWrapper, path);
 
                 var executionTimeWatch = Stopwatch.StartNew();
-                using (var reader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                try
                 {
-                    while (await reader.ReadAsync(cancellationToken))
+                    using (var reader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
                     {
-                        (int tagKey, string tagPath, string tagVR, string tagPrivateCreator, int tagLevel, int tagStatus) = reader.ReadRow(
-                            VLatest.ExtendedQueryTag.TagKey,
-                            VLatest.ExtendedQueryTag.TagPath,
-                            VLatest.ExtendedQueryTag.TagVR,
-                            VLatest.ExtendedQueryTag.TagPrivateCreator,
-                            VLatest.ExtendedQueryTag.TagLevel,
-                            VLatest.ExtendedQueryTag.TagStatus);
-
-                        results.Add(new ExtendedQueryTagStoreEntry(tagKey, tagPath, tagVR, tagPrivateCreator, (QueryTagLevel)tagLevel, (ExtendedQueryTagStatus)tagStatus));
+                        while (await reader.ReadAsync(cancellationToken))
+                        {
+                            (int tagKey, string tagPath, string tagVR, string tagPrivateCreator, int tagLevel, int tagStatus) = reader.ReadRow(
+                                VLatest.ExtendedQueryTag.TagKey,
+                                VLatest.ExtendedQueryTag.TagPath,
+                                VLatest.ExtendedQueryTag.TagVR,
+                                VLatest.ExtendedQueryTag.TagPrivateCreator,
+                                VLatest.ExtendedQueryTag.TagLevel,
+                                VLatest.ExtendedQueryTag.TagStatus);
+
+                            results.Add(new ExtendedQueryTagStoreEntry(tagKey, tagPath, tagVR, tagPrivateCreator, (QueryTagLevel)tagLevel, (ExtendedQueryTagStatus)tagStatus));
+                        }
+
+                        executionTimeWatch.Stop();
+                        Logger.LogInformation(executionTimeWatch.ElapsedMilliseconds.ToString());
```

Comment:
Should this log message have more context?

Reply:
Moved this change out of the PR